frontend/js/behaviors/core/pinboard.js

```diff
@@ -1,197 +1,246 @@
-    let marginLeft = (container.offsetWidth - (colWidth * colCount)) / (colCount - 1);
-    firstChild.classList.add('s-repositioning');
-    let marginTop = _getMarginTop(firstChild);
-    firstChild.classList.remove('s-repositioning');
-    marginTop = (typeof marginTop === 'number') ? marginTop : 60;
-    colCurrent = 0;
-    forEach(blocks, function(index, block) {
-      if (block.classList.contains('s-positioned') === false || resetPreviousPositions) {
-        // Reset the blocks height
-        block.style.height = 'auto';
-        // Work out which col to drop into and how far left this is
-        let smallestCol;
-        let smallestColIndex;
-        let leftPos;
-        if (maintainOrder) {
-          // Always maintain DOM order
-          smallestCol = cols[colCurrent];
-          smallestColIndex = colCurrent;
-          leftPos = smallestColIndex * (colWidth + marginLeft);
-          colCurrent = (colCurrent < cols.length - 1) ? colCurrent + 1 : 0;
+
+    function _setupBlocks() {
+        colCount = colCounts[A17.currentMediaQuery];
+        if (colCount) {
+            cols = [];
+            for (var i = 0; i < colCount; i++) {
+                cols.push(0);
+            }
+            active = true;
+            _positionBlocks(true);
         } else {
-          // Best position like pinterest/masonry
-          smallestCol = _minOfArray(cols);
-          smallestColIndex = cols.indexOf(smallestCol);
-          leftPos = smallestColIndex * (colWidth + marginLeft);
+            _unpositionBlocks();
+            active = false;
         }
-        // Calc and lock any image heights, make height is 4:3 ratio
-        let img = block.querySelector('img');
-        if (img && block.classList.contains('m-listing--variable-height')) {
-          let blockWidth = block.offsetWidth;
-          let maxHeight = Math.round(block.offsetWidth * 4/3);
-          let imageNativeHeightAtThisWidth = Math.round((parseInt(img.getAttribute('height')) / parseInt(img.getAttribute('width'))) * blockWidth);
-          if (imageNativeHeightAtThisWidth > maxHeight) {
-            img.parentNode.style.height = maxHeight + 'px';
-          } else {
-            img.parentNode.style.height = imageNativeHeightAtThisWidth + 'px';
-          }
+    }
+
+    function _contentAdded() {
+        _positionBlocks();
+    }
+
+    function _resized() {
+        setTimeout(function() {
+            _getColCounts(
+                document.documentElement.classList.contains(
+                    's-collection-filters-active',
+                ) && optionLayout
+                    ? optionLayout
+                    : container.className,
+            );
+            _setupBlocks();
+        }, 32);
+    }
+
+    function _showFilters() {
+        if (mediaQuery('medium+')) {
+            setTimeout(function() {
+                _getColCounts(optionLayout);
+                _setupBlocks();
+            }, 432);
```

Comment:
Do you know why these two `setTimeout()`s use these very specific timings, 32ms and 432ms?

Reply:
@zachgarwood It's just where I could get the proper timing for the load on this component. In the setup before they weren't properly evaluating the heights after the DOM was loaded and adding a bit of a delay here helps it setup and get the values it needs to execute properly without wonkyness. This is probably something we'll need to fix later with an event capture of some sort but I couldn't piggyback off of anything concrete.